typedData/typedData_test.go

```diff
@@ -0,0 +1,346 @@
+package typedData
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+var typedDataExamples = make(map[string]TypedData)
+
+// TestMain initializes test data by loading TypedData examples from JSON files.
+// It reads multiple test files and stores them in the typedDataExamples map
+// before running the tests.
+//
+// Parameters:
+// - m: The testing.M object that provides the test runner
+// Returns:
+// - None (calls os.Exit directly)
+func TestMain(m *testing.M) {
+	fileNames := []string{
+		"baseExample",
+		"example_array",
+		"example_baseTypes",
+		"example_enum",
+		"example_presetTypes",
+		"mail_StructArray",
+		"session_MerkleTree",
+		"v1Nested",
+		"allInOne",
+	}
+
+	for _, fileName := range fileNames {
+		var ttd TypedData
+		content, err := os.ReadFile(fmt.Sprintf("./tests/%s.json", fileName))
+		if err != nil {
+			panic(fmt.Errorf("fail to read file: %w", err))
+		}
+		err = json.Unmarshal(content, &ttd)
+		if err != nil {
+			panic(fmt.Errorf("fail to unmarshal TypedData: %w", err))
+		}
+
+		typedDataExamples[fileName] = ttd
+	}
+
+	os.Exit(m.Run())
+}
+
+// BMockTypedData is a helper function for benchmarks that loads a base example
+// TypedData from a JSON file.
+//
+// Parameters:
+// - b: The testing.B object used for benchmarking
+// Returns:
+// - ttd: A TypedData instance loaded from the base example file
+func BMockTypedData(b *testing.B) (ttd TypedData) {
+	b.Helper()
+	content, err := os.ReadFile("./tests/baseExample.json")
+	require.NoError(b, err)
+
+	err = json.Unmarshal(content, &ttd)
+	require.NoError(b, err)
+
+	return
+}
+
+// TestMessageHash tests the GetMessageHash function.
+//
+// It creates a mock TypedData and sets up a test case for hashing a mail message.
+// The mail message contains information about the sender and recipient, as well as the contents of the message.
+// The function then calls the GetMessageHash function with the necessary parameters to calculate the message hash.
+// If an error occurs during the hashing process, an error is reported using the t.Errorf function.
+// The expected hash value is compared with the actual hash value returned by the function.
+// If the values do not match, an error is reported using the t.Errorf function.
+//
+// Parameters:
+// - t: a testing.T object that provides methods for testing functions
+// Returns:
+// - None
+func TestGetMessageHash(t *testing.T) {
+	type testSetType struct {
+		TypedData           TypedData
+		Address             string
+		ExpectedMessageHash string
+	}
+	testSet := []testSetType{
+		{
+			TypedData:           typedDataExamples["baseExample"],
+			Address:             "0xCD2a3d9F938E13CD947Ec05AbC7FE734Df8DD826",
+			ExpectedMessageHash: "0x6fcff244f63e38b9d88b9e3378d44757710d1b244282b435cb472053c8d78d0",
+		},
+		{
+			TypedData:           typedDataExamples["example_array"],
```

Comment:
Where did the expected hahes for: example_array, v1Nested, example_enum and allInOne, come from? ( can find the others in starknet.js, but not those)

Reply:
I created a script using starknet.js were I just needed to fill the `typedData.json` file with the JSON that I wanted to check, change some variables and the revision version, and the code would return me the result I wanted. Then, I compared them with the ones returned by starknet.go until they matched.
![image](https://github.com/user-attachments/assets/875ce9d0-26fe-4103-a78f-b22c32106bf2)

Reply:
'allInOne' was created by me